api/extra.go

```diff
@@ -49,14 +55,102 @@ func ResolveDidToHandle(ctx context.Context, xrpcc *xrpc.Client, res did.Resolve
 		return "", "", fmt.Errorf("our XRPC client is authed for a different pds (%s != %s)", svc.ServiceEndpoint, xrpcc.Host)
 	}
 
-	verdid, err := comatproto.IdentityResolveHandle(ctx, xrpcc, handle)
+	verdid, err := hr.ResolveHandleToDid(ctx, handle)
 	if err != nil {
 		return "", "", err
 	}
 
-	if verdid.Did != udid {
+	if verdid != udid {
 		return "", "", fmt.Errorf("pds server reported different did for claimed handle")
 	}
 
 	return handle, svc.ServiceEndpoint, nil
 }
+
+type HandleResolver interface {
+	ResolveHandleToDid(ctx context.Context, handle string) (string, error)
+}
+
+type ProdHandleResolver struct {
+}
+
+func (dr *ProdHandleResolver) ResolveHandleToDid(ctx context.Context, handle string) (string, error) {
```

Comment:
It would make sense to me to consolidate identifier checks on `handle` here in this method, even if redundant with previous validation. We directly inject the string in to a URL below, and should ensure we aren't doing weird security things like probing our own internal/private/VPN network, even if "just" doing an HTTP GET should be safe. Eg, a malicious handle string could have username/pass, hostname (including IP addresses and port), path, and query parameters.

I don't remember if we have a "check if valid hostname" function in indigo yet, but a quick version could probably re-use golang URL verification, then ensure that none of the weird/meta TLDs like `.arpa` are used. Full list of those weird TLDs at:

https://atproto.com/specs/handle#additional-non-syntax-restrictions

Reply:
Doing full handle validation isn't a blocking thing for this PR, but if you do want to knock it out there is typescript code and example test cases here:

https://github.com/bluesky-social/atproto/blob/main/packages/identifier/tests/handle.test.ts
https://github.com/bluesky-social/atproto/blob/main/packages/identifier/src/handle.ts

Reply:
Made an issue for it with #200, dont want to block getting this to sandbox but will get it done soon